test/unit/linear.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const sharp = require('../../');
+const fixtures = require('../fixtures');
+
+describe('Linear adjustment', function () {
+  it('applies levels adjustment', function (done) {
```

Comment:
Are you able to add a test for the code path that removes/rejoins the alpha channel?